### ISSUE TITLE
ea6f5c9f0fff4f069b3060f0c21f8c8ebc69b386 unintentionally broke display of resolved IP address on HP-UX

### ISSUE THREAD

**Issue by michael-o** [curl/curl#14894]
### I did this

```
# curl http://google.de -v
* Uses proxy env variable NO_PROXY == 'localhost,.siemens.net,.siemens.com,.siemens.de,.siemens.cloud,.innomotics.net,.innomotics.com'
* timeout on name lookup is not supported
* Host google.de:80 was resolved.
* IPv6: 2a00:1450:4001:809::2003
* IPv4: 172.217.18.99
*   Trying 172.217.18.99:80...
* Connected to google.de () port 80
> GET / HTTP/1.1
> Host: google.de
```
The actual IP address is not displayed anymore.

Bisected it down to:
```
# git bisect bad
ea6f5c9f0fff4f069b3060f0c21f8c8ebc69b386 is the first bad commit
commit ea6f5c9f0fff4f069b3060f0c21f8c8ebc69b386 (HEAD)
Author: Stefan Eissing <stefan@eissing.org>
Date:   2024-08-27T12:12:37+02:00

    connect: limit update IP info

    Update IP related information at the connection and the transfer in two
    places only: once the filter chain connects and when a transfer is added
    to a connection. The latter only updates on reuse when the filters
    already are connected.

    The only user of that information before a full connect is the HAProxy
    filter. Add cfilter CF_QUERY_IP_INFO query to let it find the
    information from the filters "below".

    This solves two issues with the previous version:
    - updates where often done twice with the same info
    - happy eyeballing filter "forks" could overwrite each others
      updates before the full winner was determined.

    Closes #14699

 lib/cf-haproxy.c               | 18 ++++++++++--------
 lib/cf-https-connect.c         |  2 --
 lib/cf-socket.c                | 29 ++++++++++++++++++++++-------
 lib/cfilters.c                 | 21 ++++++++++++++++++---
 lib/cfilters.h                 | 14 ++++++++------
 lib/connect.c                  | 27 ---------------------------
 lib/connect.h                  |  3 ---
 lib/ftp.c                      |  1 -
 lib/getinfo.c                  |  7 +++----
 lib/url.c                      | 19 ++++++++++++++++---
 tests/http/test_01_basic.py    |  8 ++++++--
 tests/http/test_02_download.py |  8 ++++++--
 tests/http/test_16_info.py     | 15 +++++++++++----
 tests/http/testenv/curl.py     | 16 +++++++++++++++-
 tests/http/testenv/env.py      | 11 +++++++----
 15 files changed, 122 insertions(+), 77 deletions(-)
```

While it seems to be a visual issue only, I cannot assess whether anything else is affected.

@icing

### I expected the following

Display the actually used IP address at "Connected to ..."

### curl/libcurl version

curl 8.10.0 (ia64-hp-hpux11.31) libcurl/8.10.0 OpenSSL/3.0.15 zlib/1.3.1
Release-Date: 2024-09-11
Protocols: file http https ipfs ipns smtp smtps
Features: GSS-API HSTS HTTPS-proxy IPv6 Kerberos Largefile libz SPNEGO SSL UnixSockets

### operating system

HP-UX deblndw0 B.11.31 U ia64 HP-UX

**Comment by michael-o** [comment]
Reverted the commit on top of 8.10.0:
```
root@deblndw002x:/var/tmp/ports/work/curl ((curl-8_10_0) +)
# curl --version
curl 8.10.0-DEV (ia64-hp-hpux11.31) libcurl/8.10.0-DEV OpenSSL/3.0.15 zlib/1.3.1
Release-Date: [unreleased]
Protocols: file http https ipfs ipns smtp smtps
Features: GSS-API HSTS HTTPS-proxy IPv6 Kerberos Largefile libz SPNEGO SSL UnixSockets
root@deblndw002x:/var/tmp/ports/work/curl ((curl-8_10_0) +)
# curl http://google.de -v
* Uses proxy env variable NO_PROXY == 'localhost,.siemens.net,.siemens.com,.siemens.de,.siemens.cloud,.innomotics.net,.innomotics.com'
* timeout on name lookup is not supported
* Host google.de:80 was resolved.
* IPv6: 2a00:1450:4001:806::2003
* IPv4: 172.217.16.195
*   Trying [2a00:1450:4001:806::2003]:80...
* Immediate connect fail for 2a00:1450:4001:806::2003: Network is unreachable
*   Trying 172.217.16.195:80...
* Connected to google.de (172.217.16.195) port 80
> GET / HTTP/1.1
> Host: google.de
> User-Agent: curl/8.10.0-DEV
> Accept: */*
```

Will need to keep a local reverse patch for the time being.

**Comment by icing** [comment]
@michael-o thanks for the report. Can you give me a log with `curl -vvvv` of this?

**Comment by michael-o** [comment]
> @michael-o thanks for the report. Can you give me a log with `curl -vvvv` of this?

The fixed (one-off) or broken one?

**Comment by icing** [comment]
> > @michael-o thanks for the report. Can you give me a log with `curl -vvvv` of this?
> 
> The fixed (one-off) or broken one?

The broken one, up to the missing information, please.

**Comment by michael-o** [comment]
> > > @michael-o thanks for the report. Can you give me a log with `curl -vvvv` of this?
> > 
> > 
> > The fixed (one-off) or broken one?
> 
> The broken one, up to the missing information, please.

Here it is:
```
# curl http://google.de -vvvv -I
12:29:00.270986 [0-x] == Info: [READ] client_reset, clear readers
12:29:00.271525 [0-x] == Info: Uses proxy env variable NO_PROXY == 'localhost,.siemens.net,.siemens.com,.siemens.de,.siemens.cloud,.innomotics.net,.innomotics.com'
12:29:00.272222 [0-0] == Info: timeout on name lookup is not supported
12:29:00.273671 [0-0] == Info: Host google.de:80 was resolved.
12:29:00.273930 [0-0] == Info: IPv6: 2a00:1450:4001:806::2003
12:29:00.274160 [0-0] == Info: IPv4: 172.217.16.195
12:29:00.274373 [0-0] == Info: [SETUP] added
12:29:00.274574 [0-0] == Info: [HAPPY-EYEBALLS] created ipv6 (timeout 299997ms)
12:29:00.274880 [0-0] == Info: [HAPPY-EYEBALLS] created ipv4 (timeout 299997ms)
12:29:00.275207 [0-0] == Info: [HAPPY-EYEBALLS] ipv6 starting (timeout=299996ms)
12:29:00.275530 [0-0] == Info:   Trying [2a00:1450:4001:806::2003]:80...
12:29:00.275841 [0-0] == Info: [TCP] cf_socket_open() -> 0, fd=6
12:29:00.276127 [0-0] == Info: [TCP] local address :: port 0...
12:29:00.276423 [0-0] == Info: Immediate connect fail for 2a00:1450:4001:806::2003: Network is unreachable
12:29:00.276829 [0-0] == Info: [HAPPY-EYEBALLS] ipv6 connect -> 7, connected=0
12:29:00.277122 [0-0] == Info: [HAPPY-EYEBALLS] ipv6 done
12:29:00.277348 [0-0] == Info: [HAPPY-EYEBALLS] ipv4 starting (timeout=299996ms)
12:29:00.277660 [0-0] == Info:   Trying 172.217.16.195:80...
12:29:00.277917 [0-0] == Info: [TCP] cf_socket_open() -> 0, fd=6
12:29:00.278188 [0-0] == Info: [TCP] local address 10.64.105.134 port 57169...
12:29:00.278485 [0-0] == Info: [HAPPY-EYEBALLS] ipv4 connect -> 0, connected=0
12:29:00.278818 [0-0] == Info: [TCP] adjust_pollset, !connected, POLLOUT fd=6
12:29:00.279112 [0-0] == Info: [HAPPY-EYEBALLS] adjust_pollset -> 1 socks
12:29:00.290732 [0-0] == Info: [TCP] connected
12:29:00.290910 [0-0] == Info: [HAPPY-EYEBALLS] ipv4 connect -> 0, connected=1
12:29:00.291212 [0-0] == Info: [TCP] destroy
12:29:00.291387 [0-0] == Info: Connected to google.de () port 80
12:29:00.291726 [0-0] == Info: [TCP] send(len=74) -> 74, err=0
12:29:00.291962 [0-0] => Send header, 74 bytes (0x4a)
0000: HEAD / HTTP/1.1
0011: Host: google.de
0022: User-Agent: curl/8.10.0
003b: Accept: */*
0048:
12:29:00.292601 [0-0] == Info: [TCP] recv(len=102400) -> -1, err=81
12:29:00.292882 [0-0] == Info: Request completely sent off
12:29:00.315780 [0-0] == Info: [TCP] recv(len=102400) -> 572, err=0
12:29:00.316052 [0-0] <= Recv header, 32 bytes (0x20)
0000: HTTP/1.1 301 Moved Permanently
HTTP/1.1 301 Moved Permanently
12:29:00.316441 [0-0] == Info: [WRITE] cw_out, wrote 32 header bytes -> 32
12:29:00.316721 [0-0] == Info: [WRITE] download_write header(type=c, blen=32) -> 0
12:29:00.317027 [0-0] == Info: [WRITE] client_write(type=c, len=32) -> 0
12:29:00.317299 [0-0] <= Recv header, 33 bytes (0x21)
0000: Location: http://www.google.de/
12:29:00.317652 [0-0] == Info: [WRITE] header_collect pushed(type=1, len=33) -> 0
Location: http://www.google.de/
12:29:00.318005 [0-0] == Info: [WRITE] cw_out, wrote 33 header bytes -> 33
12:29:00.318282 [0-0] == Info: [WRITE] download_write header(type=4, blen=33) -> 0
12:29:00.318591 [0-0] == Info: [WRITE] client_write(type=4, len=33) -> 0
12:29:00.318862 [0-0] <= Recv header, 40 bytes (0x28)
0000: Content-Type: text/html; charset=UTF-8
12:29:00.319244 [0-0] == Info: [WRITE] header_collect pushed(type=1, len=40) -> 0
Content-Type: text/html; charset=UTF-8
12:29:00.319560 [0-0] == Info: [WRITE] cw_out, wrote 40 header bytes -> 40
12:29:00.319839 [0-0] == Info: [WRITE] download_write header(type=4, blen=40) -> 0
12:29:00.320145 [0-0] == Info: [WRITE] client_write(type=4, len=40) -> 0
12:29:00.320415 [0-0] <= Recv header, 245 bytes (0xf5)
0000: Content-Security-Policy-Report-Only: object-src 'none';base-uri
0040: 'self';script-src 'nonce-y_hoKc9MmP4LKjsoz1J5DA' 'strict-dynamic
0080: ' 'report-sample' 'unsafe-eval' 'unsafe-inline' https: http:;rep
00c0: ort-uri https://csp.withgoogle.com/csp/gws/other-hp
12:29:00.321696 [0-0] == Info: [WRITE] header_collect pushed(type=1, len=245) -> 0
Content-Security-Policy-Report-Only: object-src 'none';base-uri 'self';script-src 'nonce-y_hoKc9MmP4LKjsoz1J5DA' 'strict-dynamic' 'report-sample' 'unsafe-eval' 'unsafe-inline' https: http:;report-uri https://csp.withgoogle.com/csp/gws/other-hp
12:29:00.322013 [0-0] == Info: [WRITE] cw_out, wrote 245 header bytes -> 245
12:29:00.322299 [0-0] == Info: [WRITE] download_write header(type=4, blen=245) -> 0
12:29:00.322610 [0-0] == Info: [WRITE] client_write(type=4, len=245) -> 0
12:29:00.322887 [0-0] <= Recv header, 37 bytes (0x25)
0000: Date: Fri, 13 Sep 2024 10:29:00 GMT
12:29:00.323258 [0-0] == Info: [WRITE] header_collect pushed(type=1, len=37) -> 0
Date: Fri, 13 Sep 2024 10:29:00 GMT
12:29:00.323570 [0-0] == Info: [WRITE] cw_out, wrote 37 header bytes -> 37
12:29:00.323847 [0-0] == Info: [WRITE] download_write header(type=4, blen=37) -> 0
12:29:00.324155 [0-0] == Info: [WRITE] client_write(type=4, len=37) -> 0
12:29:00.324426 [0-0] <= Recv header, 40 bytes (0x28)
0000: Expires: Sun, 13 Oct 2024 10:29:00 GMT
12:29:00.324805 [0-0] == Info: [WRITE] header_collect pushed(type=1, len=40) -> 0
Expires: Sun, 13 Oct 2024 10:29:00 GMT
12:29:00.325115 [0-0] == Info: [WRITE] cw_out, wrote 40 header bytes -> 40
12:29:00.325393 [0-0] == Info: [WRITE] download_write header(type=4, blen=40) -> 0
12:29:00.325702 [0-0] == Info: [WRITE] client_write(type=4, len=40) -> 0
12:29:00.325973 [0-0] <= Recv header, 40 bytes (0x28)
0000: Cache-Control: public, max-age=2592000
12:29:00.326352 [0-0] == Info: [WRITE] header_collect pushed(type=1, len=40) -> 0
Cache-Control: public, max-age=2592000
12:29:00.326658 [0-0] == Info: [WRITE] cw_out, wrote 40 header bytes -> 40
12:29:00.326932 [0-0] == Info: [WRITE] download_write header(type=4, blen=40) -> 0
12:29:00.327239 [0-0] == Info: [WRITE] client_write(type=4, len=40) -> 0
12:29:00.327506 [0-0] <= Recv header, 13 bytes (0xd)
0000: Server: gws
12:29:00.327773 [0-0] == Info: [WRITE] header_collect pushed(type=1, len=13) -> 0
Server: gws
12:29:00.328087 [0-0] == Info: [WRITE] cw_out, wrote 13 header bytes -> 13
12:29:00.328368 [0-0] == Info: [WRITE] download_write header(type=4, blen=13) -> 0
12:29:00.328677 [0-0] == Info: [WRITE] client_write(type=4, len=13) -> 0
12:29:00.328954 [0-0] <= Recv header, 21 bytes (0x15)
0000: Content-Length: 218
12:29:00.329260 [0-0] == Info: [WRITE] header_collect pushed(type=1, len=21) -> 0
Content-Length: 218
12:29:00.329570 [0-0] == Info: [WRITE] cw_out, wrote 21 header bytes -> 21
12:29:00.329851 [0-0] == Info: [WRITE] download_write header(type=4, blen=21) -> 0
12:29:00.330156 [0-0] == Info: [WRITE] client_write(type=4, len=21) -> 0
12:29:00.330421 [0-0] <= Recv header, 21 bytes (0x15)
0000: X-XSS-Protection: 0
12:29:00.330721 [0-0] == Info: [WRITE] header_collect pushed(type=1, len=21) -> 0
X-XSS-Protection: 0
12:29:00.331025 [0-0] == Info: [WRITE] cw_out, wrote 21 header bytes -> 21
12:29:00.331298 [0-0] == Info: [WRITE] download_write header(type=4, blen=21) -> 0
12:29:00.331607 [0-0] == Info: [WRITE] client_write(type=4, len=21) -> 0
12:29:00.331876 [0-0] <= Recv header, 29 bytes (0x1d)
0000: X-Frame-Options: SAMEORIGIN
12:29:00.332213 [0-0] == Info: [WRITE] header_collect pushed(type=1, len=29) -> 0
X-Frame-Options: SAMEORIGIN
12:29:00.332526 [0-0] == Info: [WRITE] cw_out, wrote 29 header bytes -> 29
12:29:00.332802 [0-0] == Info: [WRITE] download_write header(type=4, blen=29) -> 0
12:29:00.333113 [0-0] == Info: [WRITE] client_write(type=4, len=29) -> 0
12:29:00.333386 [0-0] <= Recv header, 19 bytes (0x13)
0000: Connection: close
12:29:00.333686 [0-0] == Info: [WRITE] header_collect pushed(type=1, len=19) -> 0
Connection: close
12:29:00.333998 [0-0] == Info: [WRITE] cw_out, wrote 19 header bytes -> 19
12:29:00.334277 [0-0] == Info: [WRITE] download_write header(type=4, blen=19) -> 0
12:29:00.334592 [0-0] == Info: [WRITE] client_write(type=4, len=19) -> 0
12:29:00.334870 [0-0] <= Recv header, 2 bytes (0x2)
0000:
12:29:00.335093 [0-0] == Info: [WRITE] header_collect pushed(type=1, len=2) -> 0

12:29:00.335400 [0-0] == Info: [WRITE] cw_out, wrote 2 header bytes -> 2
12:29:00.335673 [0-0] == Info: [WRITE] download_write header(type=4, blen=2) -> 0
12:29:00.335979 [0-0] == Info: [WRITE] client_write(type=4, len=2) -> 0
12:29:00.336249 [0-0] == Info: [WRITE] xfer_write_resp(len=572, eos=0) -> 0
12:29:00.336565 [0-0] == Info: [WRITE] cw-out is notpaused
12:29:00.336829 [0-0] == Info: [WRITE] cw-out done
12:29:00.337042 [0-0] == Info: [READ] client_reset, clear readers
12:29:00.337344 [0-0] == Info: shutting down connection #0
12:29:00.337589 [0-0] == Info: [SETUP] shut down successfully
12:29:00.337844 [0-0] == Info: [HAPPY-EYEBALLS] shut down successfully
12:29:00.338118 [0-0] == Info: [TCP] cf_socket_shutdown(6)
12:29:00.338354 [0-0] == Info: [TCP] shut down successfully
12:29:00.338590 [0-0] == Info: [SETUP] close
12:29:00.338775 [0-0] == Info: [HAPPY-EYEBALLS] close
12:29:00.338994 [0-0] == Info: [TCP] cf_socket_close(6)
12:29:00.339230 [0-0] == Info: [TCP] destroy
12:29:00.339414 [0-0] == Info: [HAPPY-EYEBALLS] destroy
12:29:00.339643 [0-0] == Info: [SETUP] destroy
```

**Comment by icing** [comment]
Thanks! I made #14897 as a fix for this.

**Comment by bagder** [comment]
Closed via 50166c0de8af1aec. I accidentally did not tag the commit correctly to refer to this issue.
